src/descriptions.cpp

```diff
@@ -210,5 +212,34 @@ std::string map_data_common_t::extended_description() const
         ss << std::endl;
     }
 
+    // Flags
+    // TODO : These should be in JSON
+    const std::vector<std::pair<ter_furn_flag, std::string>> flag_descriptions = {
+        { ter_furn_flag::TFLAG_BARRICADABLE_DOOR, _( "<info>It is a door</info> that <good>can be barricaded</good>." ) },
+        { ter_furn_flag::TFLAG_BLOCK_WIND, _( "It can <good>block the effects of wind</good>." ) },
+        { ter_furn_flag::TFLAG_EASY_DECONSTRUCT, _( "It can be <good>deconstructed without tools</good>." ) },
+        { ter_furn_flag::TFLAG_FISHABLE, _( "You can <good>try to catch fish here</good>." ) },
+        { ter_furn_flag::TFLAG_FIRE_CONTAINER, _( "It can <good>safely contain a fire</good>." ) },
+        { ter_furn_flag::TFLAG_SUPPRESS_SMOKE, _( "Fires inside this <good>won't emit smoke into their surroundings</good>." ) },
+        { ter_furn_flag::TFLAG_FLAMMABLE, _( "It is <bad>flammable</bad>." ) },
+        { ter_furn_flag::TFLAG_FLAMMABLE_HARD, _( "It is <info>hard to light on fire</info>, but still possible." ) },
+        { ter_furn_flag::TFLAG_FLAT_SURF, _( "It provides a <good>clean surface</good> on which to work." ) },
+        { ter_furn_flag::TFLAG_FLAT, _( "You could <info>build or move furniture here</info>." ) },
+        { ter_furn_flag::TFLAG_HIDE_PLACE, _( "You could <info>hide inside</info>." ) },
+        { ter_furn_flag::TFLAG_PLANTABLE, _( "You could <good>plant seeds here</good>." ) },
+        { ter_furn_flag::TFLAG_ROAD, _( "You could <info>drive or skate here easily</info>." ) },
+        { ter_furn_flag::TFLAG_ROUGH, _( "It would be <bad>dangerous to walk on with bare feet</bad>." ) },
+        { ter_furn_flag::TFLAG_SHARP, _( "It could <bad>damage anything that passed through it</bad>." ) },
+        { ter_furn_flag::TFLAG_SUPPORTS_ROOF, _( "It can <info>support a roof</info>." ) }
+    };
+
+    ss << std::endl;
+
+    for( const std::pair<ter_furn_flag, std::string> &flag_pair : flag_descriptions ) {
```

Comment:
If you used a `std::map` instead of a `std::vector` of pairs, you could use `std::map::find` to look for the matching entry instead of iterating through the container manually.

Reply:
I agree 100%! I was pretty tired and just copied the way `clothing_flags_description()` did things. Thank you!

Flipping this PR to a draft so I can do this, and add a few more descriptions.